client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/AbstractKaaClient.java

```diff
@@ -233,32 +230,30 @@
 
         configurationManager = new ResyncConfigurationManager(properties);
 
-        transports.put(TransportType.BOOTSTRAP, bootstrapTransport);
+        initTransport(bootstrapTransport);
         profileTransport.setProfileManager(profileManager);
         profileTransport.setClientProperties(this.properties);
-        transports.put(TransportType.PROFILE, profileTransport);
+        initTransport(profileTransport);
         eventTransport.setEventManager(eventManager);
-        transports.put(TransportType.EVENT, eventTransport);
+        initTransport(eventTransport);
         notificationTransport.setNotificationProcessor(notificationManager);
-        transports.put(TransportType.NOTIFICATION, notificationTransport);
+        initTransport(notificationTransport);
         configurationTransport.setConfigurationHashContainer(configurationManager.getConfigurationHashContainer());
         configurationTransport.setConfigurationProcessor(configurationManager.getConfigurationProcessor());
         // TODO: this should be part of properties and provided by user during
         // SDK generation
         configurationTransport.setResyncOnly(true);
-        transports.put(TransportType.CONFIGURATION, configurationTransport);
+        initTransport(configurationTransport);
         userTransport.setEndpointRegistrationProcessor(endpointRegistrationManager);
-        transports.put(TransportType.USER, userTransport);
+        initTransport(userTransport);
         redirectionTransport.setBootstrapManager(bootstrapManager);
-        transports.put(TransportType.LOGGING, logTransport);
+        initTransport(logTransport);
         logTransport.setLogProcessor(logCollector);
+    }
 
-        for (KaaTransport transport : transports.values()) {
-            transport.setChannelManager(channelManager);
-            transport.setClientState(kaaClientState);
-        }
-
-        channelManager.setConnectivityChecker(context.createConnectivityChecker());
+    private void initTransport(KaaTransport transport) {
```

Comment:
Maybe it better to add initTransport(List<KaaTransport> transport) method with argument Arrays.asList({bootstrapTransport, profileTransport, ...})